### PR TITLE
[EI-580]Add "buildMessage" attribute to the failover and loadbalance endpoints

### PR DESCRIPTION
**Pull request by vinok88**
This attribute allows overwriting the global property
"build.message.on.failover.enable" for a single endpoint config .
Fix https://github.com/wso2/product-ei/issues/580

### CODE REVIEW

File: modules/core/src/main/java/org/apache/synapse/endpoints/LoadbalanceEndpoint.java [hunk 1]
```diff
@@ -481,5 +492,27 @@ private boolean canConnect(Member member) {
...
+
+    /**
+     * Build the message
+     * @param synCtx Synapse Context
+     */
+    private void buildMessage(MessageContext synCtx) {
+        try {
+            RelayUtils.buildMessage(((Axis2MessageContext) synCtx).getAxis2MessageContext());
+        } catch (IOException | XMLStreamException ex) {
+            String msg = "Error while building the message";
```
Comment (milindaperera) [inline modules/core/src/main/java/org/apache/synapse/endpoints/LoadbalanceEndpoint.java]: We don't need this msg variable

File: modules/core/src/main/java/org/apache/synapse/endpoints/FailoverEndpoint.java [hunk 2]
```diff
@@ -109,14 +119,14 @@ private void sendMessage(MessageContext synCtx) {
...
             }
             //preserving the payload to send next endpoint if needed
-            if(buildMessage) {
-                try {
-                    RelayUtils.buildMessage(((Axis2MessageContext) synCtx).getAxis2MessageContext());
-                } catch (IOException | XMLStreamException ex) {
-                    String msg = "Error while building the message";
-                    handleException(msg, ex);
-
+            //If buildMessage attribute available in failover config it is honoured.
```
Comment (sohaniwso2) [inline modules/core/src/main/java/org/apache/synapse/endpoints/FailoverEndpoint.java]: shall we put a multi-line comment?

File: modules/core/src/main/java/org/apache/synapse/endpoints/LoadbalanceEndpoint.java [hunk 3]
```diff
@@ -187,15 +197,16 @@ private void sendMessage(MessageContext synCtx) {
...
-                        try {
-                            RelayUtils.buildMessage(((Axis2MessageContext) synCtx).getAxis2MessageContext());
-                        } catch (IOException | XMLStreamException ex) {
-                            String msg = "Error while building the message";
-                            handleException(msg, ex);
-
+                    //If buildMessage attribute available in failover config it is honoured.
+                    //else global property is considered
+                    if (isBuildMessageAttAvailable) {
+                        if (buildMessageAtt) {
```
Comment (milindaperera) [inline modules/core/src/main/java/org/apache/synapse/endpoints/LoadbalanceEndpoint.java]: we can merge if statements

File: modules/core/src/main/java/org/apache/synapse/endpoints/FailoverEndpoint.java [hunk 4]
```diff
@@ -109,14 +119,14 @@ private void sendMessage(MessageContext synCtx) {
...
+            //else global property is considered
+            if (isBuildMessageAttAvailable) {
+                if (buildMessageAtt) {
```
Comment (milindaperera) [inline modules/core/src/main/java/org/apache/synapse/endpoints/FailoverEndpoint.java]: can merge to one if statement

File: modules/core/src/main/java/org/apache/synapse/endpoints/FailoverEndpoint.java [hunk 5]
```diff
@@ -279,4 +289,34 @@ public boolean isDynamic() {
...
+
+    /**
+     * Build the message
+     * @param synCtx Synapse Context
+     */
+    private void buildMessage(MessageContext synCtx) {
+        try {
+            RelayUtils.buildMessage(((Axis2MessageContext) synCtx).getAxis2MessageContext());
+        } catch (IOException | XMLStreamException ex) {
+            String msg = "Error while building the message";
```
Comment (milindaperera) [inline modules/core/src/main/java/org/apache/synapse/endpoints/FailoverEndpoint.java]: We don't need this variable

**Pull request merged**